### ISSUE TITLE
bug(datepicker): Missing to provide the confirmation of selected date after closing the calendar

### ISSUE THREAD

**Issue by zarend** [angular/components#23568]
#### Reproduction

Steps to reproduce:
1. Open the [Datepicker Examples](https://material.angular.io/components/datepicker/examples)
2. Go to the Basic Datepicker demo
3. Activate "Open calendar" button from any section, say, "Single date selection"
4. Apply any date by selecting one from the datepicker
5. Listen the audio feedback
 
#### Expected Behavior
It should provide audio feedback with selected date shown in the edit box

#### Actual Behavior
It is not providing the audio feedback of selected date shown on the edit box

#### Environment
OS| Browser | Screen reader
ChromeOS | Google Chrome 91.0.4472.167 | ChromeVox
macOS Catalina | Google Chrome 92.0.4515.131 | VoiceOver
Windows 10 | Google Chrome 92.0.4515.131 | NVDA/JAWS
Windows 10 | Firefox 90.0.2 (64-bit) | NVDA

##### Additional Notes
This is copied from an [internal bug report](http://b/195600323) found during a11y testing.

Fix is to announce the date on selection.

**Comment by zarend** [comment]
@jelbourn , I thought of 3 approaches for this

1. Use confirmation buttons
2. Focus the input after selecting a date like the Lightning React datepicker
3. Add the current day to the aria-label on the datepicker trigger `aria-label="Open calendar, Dec 10 2021"` Like the [Date Picker Dialog Example w3 example](https://www.w3.org/TR/wai-aria-practices-1.2/examples/dialog-modal/datepicker-dialog.html)

1 seems like the safest option for having the most accessible experience
2 I'm leaning towards this approach. The [dialog spec aria pattern](https://www.w3.org/TR/wai-aria-practices-1.2/#keyboard-interaction-7) allows focusing a different element when "The task completed in the dialog is directly related to a subsequent step in the work flow."
3. keeps focus on the element that opens the datpicker, but could be tricky with internationalization

WDYT?

**Comment by jelbourn** [comment]
Yeah, option 1 here is the most conventional approach. Since we already support apply/cancel buttons for single date selection, maybe this is something we can just add to the guidance in the documentation.
